pkg/scheduler/internal/cache/cache.go

```diff
@@ -501,17 +501,19 @@ func (cache *cacheImpl) AddPod(pod *v1.Pod) error {
 	currState, ok := cache.podStates[key]
 	switch {
 	case ok && cache.assumedPods.Has(key):
+		// When assuming, we've already added the Pod to cache,
+		// Just update here to make sure the Pod's status is up-to-date.
+		if err = cache.updatePod(currState.pod, pod); err != nil {
```

Comment:
`removePod` -> `addPod` seems always right, but maybe we can optimize here by only updating the podInfo, which performs better, WDYT? https://github.com/kubernetes/kubernetes/blob/8415ae647d2c433c89910a0e677094e3a20ffb2b/pkg/scheduler/framework/types.go#L370-L376

By the way, do we use the status somewhere?

Reply:
> removePod -> addPod seems always right, but maybe we can optimize here by only updating the podInfo, which performs better, WDYT?

sounds good to me.

I think we can optimize `updatePod` method like: 
- when `oldPod.Spec.NodeName == newPod.Spec.NodeName`, updates only PodInfo in NodeInfo. (Also, we probably need to create new methods on `framework.NodeInfo` like `UpdatePod`/`UpdatePodInfo`.)
- otherwise, deletePod -> addPod. (the current behavior.)

Let me create a separate PR for that.

> By the way, do we use the status somewhere?

Not sure. I found this issue because my company uses the status in the plugin.